### PR TITLE
Congestion visualization

### PR DESCRIPTION
**Pull request by yasushisakai**
Gives you this:

c7eea379-8b4f-4579-b11f-5eac36c7686b

### CODE REVIEW

**Comment by agrignard** [comment]
@yasushisakai it sounds god and a lot fo modification in the code. How does it behave in term of performance?

I would make the same remark as I did for this PR https://github.com/CityScope/CS_Cooper-Hewitt/pull/104. Let's first focus on having a release of the plan A today and then we can go to A+ and even more by pushing some optimization and new rendering (but it needs to be validated by everyone first) but not before having a final release A

As I said from the beginning if we have to surimpose a heatmap on the agent that means we have miss something from the beginning in the design , if you think about a heatmap it is supposed to be an abstraction/aggregation of something and in our case we show the abstraction (the heatmap) and this something (the agent). I am afraid ti will be to much and or redundant (I mean we can already see that there is lot of car no?) but we have to test on the table for sure, by running this branche.

So I wouldn't  merge this one either  to be sure  not to introduce last minute debugging (we cannot afford this anymore)

**Comment by agrignard** [comment]
Just by testing it quickly I see a regression in the rendering of the network (just press 'n') and you will see.
screen shot 2018-11-26 at 07 20 01


What I propose is to iterate in this branch but it needs more test to be sure it's stable . Maybe we have to think of having different computer running different branches to validate them.

Let's talk this about this today (for the design it needs to be discuss) I like the programmatic solution but I am not 100% sure of the final  rendering

**Comment by yasushisakai** [comment]
Yes, indeed, this needs careful scrutiny

**Comment by yasushisakai** [comment]
Not to impose anything, but 469d929 fixes the thick ‘n’ show network function.

**Comment by agrignard** [comment]
Cool 'n' is working now. 

Sometime some black rectangle are displayed and then we don't see anymore the background 
I don't think we want this, maybe not display the rectangle if black?

screen shot 2018-11-26 at 11 45 06

File: ABMobility/RoadNetwork.pde [hunk 1]
```diff
@@ -113,6 +118,22 @@ public class RoadNetwork {
...
+      Node srcNode, 
+      Node toNode) { 
+
+     return edgeManager.updateEdge(agent, edge, srcNode, toNode); 
+
+  }
+
+  public void drawEdges(PGraphics p) {
+    p.pushStyle();
+      edgeManager.draw(p);
```
Comment (aberke) [inline ABMobility/RoadNetwork.pde]: fix indentation plz

File: ABMobility/RoadNetwork.pde [hunk 2]
```diff
@@ -113,6 +118,22 @@ public class RoadNetwork {
...
   }
+
+  public NetworkEdge updateEdge(
+      Agent agent, 
+      NetworkEdge edge, 
+      Node srcNode, 
+      Node toNode) { 
+
+     return edgeManager.updateEdge(agent, edge, srcNode, toNode); 
+
```
Comment (aberke) [inline ABMobility/RoadNetwork.pde]: remove extra newline plz

**Review by aberke — APPROVED** [review]
This looks good to me - nice job!  I made some stylistic nitpicks in the comments.

**Comment by yasushisakai** [comment]
I’ll take a look in the morning, my mouth hurts. (took a tooth) 🔧  👄 🔨

**Comment by yasushisakai** [comment]
Something has changed to the master 469d929 can no longer be applied...

I tried to make another pull request branching from 469d929 and merge it with the current master, but as I solve the conflicts, agents will not move anymore.

**Pull request closed without merging**